pkg/controllers/managementuser/secret/secret.go

```diff
@@ -115,6 +119,9 @@ func (n *NamespaceController) sync(key string, obj *corev1.Namespace) (runtime.O
 					continue
 				}
 				namespacedSecret := getNamespacedSecret(secret, obj.Name)
+				if _, err := n.clusterSecretsLister.Get(namespacedSecret.Namespace, namespacedSecret.Name); err == nil {
+					continue
+				}
 				logrus.Infof("Creating secret [%s] into namespace [%s]", namespacedSecret.Name, obj.Name)
 				_, err := n.clusterSecretsClient.Create(namespacedSecret)
 				if err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
Shouldn't there be a continue in this block as well, instead of a return?

Reply:
That was the existing logic, I didn't want to change anything except the specific issue of not checking for the existence of the secret.